tachyon/math/matrix/matrix_utils.h

```diff
@@ -192,6 +193,41 @@ std::vector<Eigen::Block<Derived>> SplitMat(size_t num_chunks,
   });
 }
 
+template <typename ExtField, typename Derived>
+std::vector<ExtField> DotExtPowers(const Eigen::MatrixBase<Derived>& mat,
+                                   const ExtField& base) {
+  using F = typename ExtensionFieldTraits<ExtField>::BaseField;
+  using PackedField = typename PackedFieldTraits<F>::PackedField;
+  using ExtendedPackedField =
+      typename ExtendedPackedFieldTraits<ExtField>::ExtendedPackedField;
+  size_t rows = static_cast<size_t>(mat.rows());
+  size_t packed_n = PackedField::N;
+  std::vector<ExtendedPackedField> packed_ext_powers =
+      ExtField::GetExtendedPackedPowers(
+          base, ((mat.cols() + packed_n - 1) / packed_n) * packed_n);
+  std::vector<ExtField> ret(rows);
```

Comment:
How about only reserve here?

Reply:
I can just use `CreateVectorParallel`! I'll change that.